internal/audit/kafka/tls.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2021-2023 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package kafka
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/cerbos/cerbos/internal/observability/logging"
+	"go.uber.org/zap"
+)
+
+func NewTLSConfig(ctx context.Context, reloadInterval time.Duration, insecureSkipVerify bool, caPath, certPath, keyPath string) (*tls.Config, error) {
+	if caPath == "" {
+		return nil, errors.New("CA path cannot be empty")
+	}
+
+	if certPath != "" && keyPath == "" || certPath == "" && keyPath != "" {
+		return nil, errors.New("certPath and keyPath must both be empty or both be non-empty")
+	}
+
+	caCert, err := os.ReadFile(caPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read CA certificate: %w", err)
+	}
+
+	caCertPool := x509.NewCertPool()
+	if !caCertPool.AppendCertsFromPEM(caCert) {
+		return nil, fmt.Errorf("failed to append CA certificate")
+	}
+
+	// #nosec G402
+	tlsConfig := &tls.Config{
+		RootCAs:            caCertPool,
+		ClientCAs:          caCertPool,
```

Comment:
I don't think we need to set `ClientCAs` - my understanding is that's just for use on the server end of the connection.

>ClientCAs defines the set of root certificate authorities that servers use if required to verify a client certificate by the policy in ClientAuth.